rust/lazy/src/lazy_rule.rs

```diff
@@ -0,0 +1,142 @@
+use std::{path::Path, str::FromStr};
+
+use crate::error::{LazyError, Result};
+
+#[derive(Debug, PartialEq, Clone)]
+pub enum LazyStrategy {
+    SMUDGE,
+    POINTER,
+}
+
+impl FromStr for LazyStrategy {
+    type Err = LazyError;
+
+    fn from_str(s: &str) -> Result<Self> {
+        let lower_s = s.to_ascii_lowercase();
+        match lower_s.as_str() {
+            "smudge" => Ok(LazyStrategy::SMUDGE),
+            "pointer" => Ok(LazyStrategy::POINTER),
+            "s" => Ok(LazyStrategy::SMUDGE),
+            "p" => Ok(LazyStrategy::POINTER),
+            _ => Err(LazyError::InvalidStrategy(format!(
+                "{s}, supported strategy is smudge(s) or pointer(p)"
+            ))),
+        }
+    }
+}
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct LazyRule {
+    pub path: String,
+    pub strategy: LazyStrategy,
+}
+
+impl FromStr for LazyRule {
+    type Err = LazyError;
+
+    fn from_str(s: &str) -> Result<Self> {
+        let split = s.trim().split_once(' ').ok_or_else(|| {
+            LazyError::InvalidRule(format!("failed to parse strategy from {s:?}"))
+        })?;
+
+        let strategy: LazyStrategy = split.0.parse()?;
+        let path = split
+            .1
+            .trim()
+            .trim_matches(std::path::MAIN_SEPARATOR)
+            .to_owned();
+
+        Ok(LazyRule { path, strategy })
+    }
+}
+
+impl LazyRule {
+    pub fn match_path_prefix(&self, path: impl AsRef<Path>) -> Result<bool> {
+        let path = path
```

Comment:
I am wondering if it would be better to use a prebuilt glob matching library for this, e.g. https://docs.rs/globset/latest/globset/.  This would give us more flexibility and perhaps be easier / more reliable to work with.

Reply:
I thought about glob matching, but haven't figured out the precedence between the below two rules applying to file `foo/bar/data1`:
```
smudge foo/bar/data*
pointer f*/b*/data1
```
So for simplicity I didn't implement glob matching.

Reply:
I think we can get more feedback from users and product during trial of this feature.

Reply:
Fair.  That makes sense.